ceremonies/src/lib.rs

```diff
@@ -384,55 +412,66 @@ impl<T: Trait> Module<T> {
         for cid in cids.iter() {
             let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index();
             let pcount = <ParticipantCount>::get((cid, cindex));
+            let ecount = <EndorseesCount>::get((cid, cindex));
 
+            let mut bootstrappers =
+                Vec::with_capacity(<encointer_currencies::Module<T>>::bootstrappers(cid).len());
             let mut reputables = Vec::with_capacity(pcount as usize);
+            let mut endorsees = Vec::with_capacity(ecount as usize);
             let mut newbies = Vec::with_capacity(pcount as usize);
 
             // TODO: upfront random permutation
             for p in 1..=pcount {
                 let participant = <ParticipantRegistry<T>>::get((cid, cindex), &p);
                 if Self::participant_reputation((cid, cindex), &participant)
                     == Reputation::UnverifiedReputable
-                    || <encointer_currencies::Module<T>>::bootstrappers(cid).contains(&participant)
                 {
                     reputables.push(participant);
+                } else if <encointer_currencies::Module<T>>::bootstrappers(cid)
+                    .contains(&participant)
+                {
+                    bootstrappers.push(participant);
+                } else if <Endorsees<T>>::contains_key((cid, cindex), &participant) {
+                    endorsees.push(participant)
                 } else {
                     newbies.push(participant);
                 }
             }
-            let mut n = reputables.len();
-            n += min(newbies.len(), n / 4);
+            let mut n = bootstrappers.len() + reputables.len();
+            n += min(newbies.len(), n / 3);
+
+            let mut all_participants = bootstrappers
+                .into_iter()
+                .chain(reputables.into_iter())
+                .chain(endorsees.into_iter())
+                .chain(newbies.into_iter());
+
             let n_meetups = n / 12 + 1;
```

Comment:
this is not the same as `ceil(n/12)` ! I'd rather have `ceil()`. If we have 12 participants, we want one meetup, not two

Reply:
As we're already editing this line, could you fix #4 at the same time?